src/source/Signaling/LwsApiCalls.c

```diff
@@ -1468,6 +1479,186 @@ STATUS connectSignalingChannelLws(PSignalingClient pSignalingClient, UINT64 time
     return retStatus;
 }
 
+STATUS joinStorageSessionLws(PSignalingClient pSignalingClient, UINT64 time)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    UNUSED_PARAM(time);
+
+    PRequestInfo pRequestInfo = NULL;
+    CHAR url[MAX_URI_CHAR_LEN + 1];
+    CHAR paramsJson[MAX_JSON_PARAMETER_STRING_LEN];
+    PLwsCallInfo pLwsCallInfo = NULL;
+    PCHAR pResponseStr;
+    UINT32 resultLen;
+
+    CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
+    CHK(pSignalingClient->channelEndpointWebrtc[0] != '\0', STATUS_INTERNAL_ERROR);
+
+    // Create the API url
+    STRCPY(url, pSignalingClient->channelEndpointWebrtc);
+    STRCAT(url, JOIN_STORAGE_SESSION_API_POSTFIX);
+
+    // Prepare the json params for the call
+    if (pSignalingClient->pChannelInfo->channelRoleType == SIGNALING_CHANNEL_ROLE_TYPE_VIEWER) {
+        SNPRINTF(paramsJson, ARRAY_SIZE(paramsJson), SIGNALING_JOIN_STORAGE_SESSION_VIEWER_PARAM_JSON_TEMPLATE,
+                 pSignalingClient->channelDescription.channelArn, pSignalingClient->clientInfo.signalingClientInfo.clientId);
+    } else {
+        SNPRINTF(paramsJson, ARRAY_SIZE(paramsJson), SIGNALING_JOIN_STORAGE_SESSION_MASTER_PARAM_JSON_TEMPLATE,
+                 pSignalingClient->channelDescription.channelArn);
+    }
+
+    // Create the request info with the body
+    CHK_STATUS(createRequestInfo(url, paramsJson, pSignalingClient->pChannelInfo->pRegion, pSignalingClient->pChannelInfo->pCertPath, NULL, NULL,
+                                 SSL_CERTIFICATE_TYPE_NOT_SPECIFIED, pSignalingClient->pChannelInfo->pUserAgent,
+                                 SIGNALING_SERVICE_API_CALL_CONNECTION_TIMEOUT, SIGNALING_SERVICE_API_CALL_COMPLETION_TIMEOUT,
+                                 DEFAULT_LOW_SPEED_LIMIT, DEFAULT_LOW_SPEED_TIME_LIMIT, pSignalingClient->pAwsCredentials, &pRequestInfo));
+
+    // createRequestInfo does not have access to the getCurrentTime callback, this hook is used for tests.
+    if (pSignalingClient->signalingClientCallbacks.getCurrentTimeFn != NULL) {
+        pRequestInfo->currentTime =
+            pSignalingClient->signalingClientCallbacks.getCurrentTimeFn(pSignalingClient->signalingClientCallbacks.customData);
+    }
+
+    checkAndCorrectForClockSkew(pSignalingClient, pRequestInfo);
+
+    CHK_STATUS(createLwsCallInfo(pSignalingClient, pRequestInfo, PROTOCOL_INDEX_HTTPS,
+                                 &pLwsCallInfo)); //!< TBD. Accroding to the design document, the prefix of url will be webrtc://
+
+    if (pSignalingClient->mediaStorageConfig.storageStatus) {
+        pSignalingClient->diagnostics.joinSessionToOfferRecvTime = GETTIME();
```

Comment:
Should we also set `pSignalingClient->diagnostics.joinSessionCallTime` to the current time here? I don't think that one is currently set anywhere.

Reply:
Seems to be set ok, in my local logs I see:
```
2023-10-18 03:47:44.983 PROFILE executeJoinStorageSessionState(): [Join Session call] Time taken: 843 ms
```

And it prints this:
```
    PROFILE_CALL_WITH_T_OBJ(retStatus = joinStorageSession(pSignalingClient, time), pSignalingClient->diagnostics.joinSessionCallTime,
                            "Join Session call");
```

Which is the `joinSessionCallTime`.

I think the current calculation is correct for `joinSessionToOfferRecvTime ` which stores the time immediately before calling join session then computes it as follows:

```
            pSignalingClient->diagnostics.joinSessionToOfferRecvTime =
                pSignalingClient->offerReceivedTime - pSignalingClient->diagnostics.joinSessionToOfferRecvTime;
```

Which looks like the right thing to do.  Are you seeing something strange in your logs for join session call time?

Reply:
Ah ok, I misunderstood what the `joinSessionCallTime` is tracking - thought it was the current time when JoinSession is called, but it's the time take until it's called, and now I see it being set there with the `PROFILE_CALL_WITH_T_OBJ`. Looks good!